stdlib/public/Concurrency/Actor.cpp

```diff
@@ -364,22 +366,40 @@ const char *__swift_runtime_env_useLegacyNonCrashingExecutorChecks() {
 
 // Done this way because of the interaction with the initial value of
 // 'unexpectedExecutorLogLevel'
-bool swift_bincompat_useLegacyNonCrashingExecutorChecks() {
-  bool legacyMode = __swift_bincompat_useLegacyNonCrashingExecutorChecks();
+swift_task_is_current_executor_flag swift_bincompat_useLegacyNonCrashingExecutorChecks() {
+  swift_task_is_current_executor_flag options =
+      __swift_bincompat_useLegacyNonCrashingExecutorChecks();
 
   // Potentially, override the platform detected mode, primarily used in tests.
   if (const char *modeStr =
           __swift_runtime_env_useLegacyNonCrashingExecutorChecks()) {
+    SWIFT_TASK_DEBUG_LOG("executor checking: SWIFT_IS_CURRENT_EXECUTOR_LEGACY_MODE_OVERRIDE = '%s'", modeStr);
     if (strcmp(modeStr, "nocrash") == 0 ||
         strcmp(modeStr, "legacy") == 0) {
-      return true;
+        SWIFT_TASK_DEBUG_LOG("executor checking: SWIFT_IS_CURRENT_EXECUTOR_LEGACY_MODE_OVERRIDE = %s => NONE; options = %d",
+                           modeStr, options);
+        options = swift_task_is_current_executor_flag(
+          swift_task_is_current_executor_flag::None);
+    }
+
+    if (strcmp(modeStr, "swift62") == 0) {
```

Comment:
Major bikeshedding: environment variables don't mind dots, should this be `swift6.2` instead?

Reply:
yep, sure

Reply:
done!